internal/metric/prometheus.go

```diff
@@ -71,6 +71,11 @@ func capturePrometheusMetric(ctx context.Context, log logr.Logger, m *redskyv1be
 	// If we got NaN, it might be a Pushgateway metric that won't have a value unless we query from the scrape time
 	if math.IsNaN(value) && lastScrapeTime.After(completionTime) {
 		log.Info("Retrying Prometheus query to include final scrape", "lastScrapeTime", lastScrapeTime, "completionTime", completionTime)
+		if time.Now().Before(lastScrapeTime) {
```

Comment:
Unsure if this is the best approach or if we should change `checkReady` to return a capture error with an offset RetryAfter?